### PR TITLE
Make URL handling consistent between links and images

### PR DESCRIPTION
**Pull request by aprotim**
**Marked version:**

b6d5a6745d0b0ce1ce0cd30f5e29fb44f28d68c4

**Markdown flavor:** all

## Expectation

With sanitization on, I'd expect 

`[Link](javascript:alert('xss'))`
`Image)`

To result in

`<p>Link`
`Image</p>`

## Result

`<p>Link
Image</p>`

## What was attempted

`marked("[Link](javascript:alert('xss'))\nImage)", {sanitize: true})`

## Contributor

- [x] Test(s) exist to ensure functionality and minimize regression (if no tests added, list tests covering this PR); or,
- [ ] no tests required for this PR.
- [ ] If submitting new feature, it has been documented in the appropriate places.

## Committer

In most cases, this should be a different person than the contributor.

- [ ] Draft GitHub release notes have been updated.
- [ ] CI is green (no forced merge required).
- [ ] Merge PR

### CODE REVIEW

**Comment by UziTech** [comment]
I think we are trying to get rid of the `sanitize` option. Sanitizing can be very difficult to maintain and there are other packages that do a great job.

https://github.com/markedjs/marked/issues/1232#issuecomment-382603925

**Comment by aprotim** [comment]
Is the rename of this PR sufficient to address that concern, @styfle?

**Comment by aprotim** [comment]
I'm not super familiar with the flow here, so just making sure there's nothing else I'm supposed to do before these PRs get merged?

**Comment by styfle** [comment]
@aprotim We need 2 approvals before we can merge this. I approved, so one other person will need to approve. @UziTech expressed some reservations so we'll see what the rest of the team thinks. There's nothing you need to do besides wait 👍

File: test/new/images.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+---
+sanitize: true
+---
+[URL](javascript:alert)
```
Comment (Feder1co5oave) [inline test/new/images.md]: These are links, not images. `!` are missing

**Pull request merged**